### DISCUSSION TITLE
Feature request: Do not dismiss approvals from code-owners whose files haven't changed from revision to revision.

### DISCUSSION THREAD

**Discussion by blghns** [community/community#38806] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Any commit dismissing ALL approvals is not productive. Often, changes are small and limited to one code owner's area. Auto dismiss should dismiss approval of that granular code area owner, and not others.

Would be efficient if the auto dismissal only dismissed the code owner approvals whose files are touched.

This is a similar mechanism to when a pr commits a merge request from a main branch and doesn't dismiss reviews if there were no merge conflicts.

Today the only workaround for this is creating separate branches per code owner. While it benefits cross-team collaboration, it is problematic for git history; for a given feature this misses showing the complete changeset.

**Comment by ossanna16** [comment]
Hi there @blghns and welcome to our community! Thank you for the feedback.

**Comment by gtbuchanan** [comment]
We just migrated to GitHub and ran into this problem. It's going to be a big pain for us since we just implemented code owners. It looks like [this issue has already been resolved in GitLab](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/90578).

**Reply by blghns** [reply]
@gtbuchanan that's pretty cool! Unfortunately, I am not in a position to switch my org to use GitLab. I hope this gets enough traction here for GitHub to implement it as a feature someday!

**Comment by Balvajs** [comment]
A small shameless plug, we were facing this exact issue in my previous company, so I wrote this GH action for removing stale code reviews based on CODEOWNERS: https://github.com/Balvajs/dismiss-stale-reviews
We have been using it internally for over a year, and it worked without issues. The only downside is that it can't work with force-pushes and falls back to "dismissing all reviews" if someone force-pushes to the PR branch.

**Reply by blghns** [reply]
Wow, this is fantastic thank you for sharing!

**Comment by bigcakes** [comment]
This would be very beneficial when you have sweeping changes that have conflicts but those conflicts only affect one code owner, dealing with that frustration right now

**Comment by brt-junwuzhang** [comment]
Please, if there's any way to implement this features, that would be awesome. This could speed up development by a LOT, and it makes sense to do so.
